### PR TITLE
Fix: Correct document for dashboards assistant

### PR DESCRIPTION
**Pull request by yuye-aws**
### Description
1. Correct API from create agent to flow framework provision
2. Add guide on system index `.plugins-ml-config`

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/8733

### Version
2.18

### Frontend features
_If you're submitting documentation for an OpenSearch Dashboards feature, add a video that shows how a user will interact with the UI step by step. A voiceover is optional._ 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by yuye-aws** [comment]
@kolchfa-aws Can you help backport this PR to 2.18. It fixes existing 2.18 documentation.

File: _dashboards/dashboards-assistant/alert-insight.md [hunk 1]
```diff
@@ -197,6 +197,9 @@ POST /.plugins-ml-config/_doc/os_insight
 The created `os_insight` agent provides alert insights related to OpenSearch cluster metrics. For insights about alerts unrelated to OpenSearch cluster metrics, you need to register an agent with [this template](https://github.com/opensearch-project/flow-framework/blob/2.x/sample-templates/create-knowledge-base-alert-agent.json) and change the agent name to `KB_For_Alert_Insight`.
 {: .note}
 
+This example shows a system index. In security-enabled domains, only super admins have permission to execute this code. For information about making super admin calls, see the [System indexes]({{site.url}}{{site.baseurl}}/security/configuration/system-indices/) guide. For access permission, contact your IT administrator.
```
Comment (kolchfa-aws) [inline _dashboards/dashboards-assistant/alert-insight.md:200]:
```suggestion
This example demonstrates a system index. In security-enabled domains, only superadmins have permissions to execute this code. For information about making superadmin calls, see [System indexes]({{site.url}}{{site.baseurl}}/security/configuration/system-indices/). For access permissions, contact your system administrator.
```

File: _dashboards/dashboards-assistant/data-summary.md [hunk 2]
```diff
@@ -237,6 +238,9 @@ POST /.plugins-ml-config/_doc/os_data2summary
 ```
 {% include copy-curl.html %}
 
+This example shows a system index. In security-enabled domains, only super admins have permission to execute this code. For information about making super admin calls, see the [System indexes]({{site.url}}{{site.baseurl}}/security/configuration/system-indices/) guide. For access permission, contact your IT administrator.
```
Comment (kolchfa-aws) [inline _dashboards/dashboards-assistant/data-summary.md:241]:
```suggestion
This example demonstrates a system index. In security-enabled domains, only superadmins have permissions to execute this code. For information about making superadmin calls, see [System indexes]({{site.url}}{{site.baseurl}}/security/configuration/system-indices/). For access permissions, contact your system administrator.
```

File: _dashboards/dashboards-assistant/suggest-anomaly-detector.md [hunk 3]
```diff
@@ -55,6 +55,9 @@ POST /.plugins-ml-config/_doc/os_suggest_ad
 ```
 {% include copy-curl.html %}
 
+This example shows a system index. In security-enabled domains, only super admins have permission to execute this code. For information about making super admin calls, see the [System indexes]({{site.url}}{{site.baseurl}}/security/configuration/system-indices/) guide. For access permission, contact your IT administrator.
```
Comment (kolchfa-aws) [inline _dashboards/dashboards-assistant/suggest-anomaly-detector.md:58]:
```suggestion
This example demonstrates a system index. In security-enabled domains, only superadmins have permissions to execute this code. For information about making superadmin calls, see [System indexes]({{site.url}}{{site.baseurl}}/security/configuration/system-indices/). For access permissions, contact your system administrator.
```

File: _dashboards/dashboards-assistant/text-to-visualization.md [hunk 4]
```diff
@@ -209,6 +209,9 @@ POST /.plugins-ml-config/_doc/os_text2vega_with_instructions
 ```
 {% include copy-curl.html %}
 
+This example shows a system index. In security-enabled domains, only super admins have permission to execute this code. For information about making super admin calls, see the [System indexes]({{site.url}}{{site.baseurl}}/security/configuration/system-indices/) guide. For access permission, contact your IT administrator.
```
Comment (kolchfa-aws) [inline _dashboards/dashboards-assistant/text-to-visualization.md:212]:
```suggestion
This example demonstrates a system index. In security-enabled domains, only superadmins have permissions to execute this code. For information about making superadmin calls, see [System indexes]({{site.url}}{{site.baseurl}}/security/configuration/system-indices/). For access permissions, contact your system administrator.
```

**Review by kolchfa-aws — APPROVED** [review]
Thank you, @yuye-aws!

**Pull request merged**